src/Evolution/Systems/GrMhd/ValenciaDivClean/PrimitiveFromConservative.cpp

```diff
@@ -54,13 +55,15 @@ void PrimitiveFromConservative<OrderedListOfPrimitiveRecoverySchemes,
           const Scalar<DataVector>& tilde_d,
           const Scalar<DataVector>& tilde_tau,
           const tnsr::i<DataVector, 3, Frame::Inertial>& tilde_s,
+          const Scalar<DataVector>& tilde_electron_d,
           const tnsr::I<DataVector, 3, Frame::Inertial>& tilde_b,
           const Scalar<DataVector>& tilde_phi,
           const tnsr::ii<DataVector, 3, Frame::Inertial>& spatial_metric,
           const tnsr::II<DataVector, 3, Frame::Inertial>& inv_spatial_metric,
           const Scalar<DataVector>& sqrt_det_spatial_metric,
           const EquationsOfState::EquationOfState<true, ThermodynamicDim>&
               equation_of_state) noexcept {
+  get(*electron_fraction) = get(tilde_electron_d) / get(tilde_d);
```

Comment:
This is called after imposing a floor on tilde_d, I assume?

Reply:
Yes, the ordering is handled in the action list of the GRMHD executable.